### PR TITLE
[JENKINS-51391] Adjust error telemetry API to JEP-308

### PR DESCRIPTION
**Pull request by mandie722**
[JENKINS-51391](https://issues.jenkins-ci.org/browse/JENKINS-51391)

I added a custom error handler to standardize the format of the error responses.

### CODE REVIEW

File: services/src/app.js
Comment (batmat) [inline services/src/app.js]: AFAICT, that is adding the response designed in the spec, but not checking the test actually receives this?

I tested locally using your fix, and adding the following inside the `.catch(err => {` block seems to do:

```javascript
assert.ok(err.error);
assert.ok(err.error.status);
assert.equal(err.error.status, 'ERROR');
```

File: services/src/app.js
Comment (batmat) [inline services/src/app.js]: AIUI, re-reading https://expressjs.com/en/guide/error-handling.html, we are missing the block:

```js
if (res.headersSent) {
  return next(err);
}
```

And if you agree, that would allow us to get rid of the eslint-disable above.

File: services/src/app.js
Comment (batmat) [inline services/src/app.js]: It took me a bit of time to read this correctly, I think adding parenthesis would improve a bit readability (and also.

```js
err.statusCode = (err.code ? err.code : 500);
```

?

**Review by batmat — APPROVED** [review]
\o/

**Comment by batmat** [comment]
Not merging yet mainly for the comment about `if (res.headersSent)`, but the rest LGTM and can be addressed in followups. The addition here already greatly improves the _statu quo_.

Congrats @mandie722 !

File: services/src/app.js
Comment (mandie722) [reply services/src/app.js]: Checking this actually will be my next task

File: services/src/app.js
Comment (mandie722) [reply services/src/app.js]: I was going back and forth on this, I can add this back in.

File: services/src/app.js
Comment (mandie722) [reply services/src/app.js]: Agree, fixing

**Review by batmat — APPROVED** [review]

**Comment by batmat** [comment]
@mandie722 \o/. So FWIW, in that situation (currently "PR building" the PR):
image

i.e. when everything is approved, but the CI is still building, we wait for it to finish and be green, and then anyone can merge.

**Comment by batmat** [comment]
I suspect https://github.com/jenkins-infra/evergreen/pull/109 will need a fix once it's merged. Going to first merge here Mandie's PR, then will address my https://github.com/jenkins-infra/evergreen/pull/109 following.

**Comment by batmat** [comment]
It passed all the tests, but failed for unrelated reason because metrics 4.x plugin now depends on `variant`, and we use hardcoded latest as long as #105 is not in.

going to merge this, as anyway this is something also failing on master, and will fix.

**Pull request merged**
